### PR TITLE
Switch to proving NimbusId assignements in container chains

### PR DESCRIPTION
**Pull request by nanocryk**
To check blocks are authored by the proper collator we need to know which NimbusIds assigned collators are using. Knowing their orchestrator AccountId however is not useful. For that reason this PR changes the proof to provide assignment data over NimbusId. Since the storage used depends on the session id, we include the session id storage in the proof.

### CODE REVIEW

**Comment by tmpolaczyk** [comment]
Should we change the `authority_assignment_for_session` storage so that it doesn't depend on the session?

**Comment by girazoki** [comment]
I think if we decided to go this route its fine. Demonstrating an additional storage item does not add that much overhead, the only thing potentially worrying is that this adds more dependency with pallet indices and storages.

It also gives us more flexibility in case we want to store more than 2 session keys.

**Review by girazoki — APPROVED** [review]
I think it's good, once we fix the tests for zombienet. You probably only have to change the accountId for the nimbusId in the verification

**Review by girazoki — APPROVED** [review]

**Review by girazoki — APPROVED** [review]

**Review by girazoki — APPROVED** [review]

**Review by girazoki — APPROVED** [review]

File: test/suites/para/test_tanssi_containers.ts [hunk 1]
```diff
@@ -82,7 +82,7 @@ describeSuite({
       title: "Test assignation did not change",
       test: async function () {
         const currentSession = (await paraApi.query.session.currentIndex()).toNumber();
-        expect(currentSession).to.be.equal(0);
+        // expect(currentSession).to.be.equal(0);
```
Comment (girazoki) [inline test/suites/para/test_tanssi_containers.ts:85]:
```suggestion
```

File: test/suites/para/test_tanssi_containers.ts [hunk 2]
```diff
@@ -69,7 +69,7 @@ describeSuite({
       title: "Test Tanssi assignation is correct",
       test: async function () {
         const currentSession = (await paraApi.query.session.currentIndex()).toNumber();
-        expect(currentSession).to.be.equal(0);
+        // expect(currentSession).to.be.equal(0);
```
Comment (girazoki) [inline test/suites/para/test_tanssi_containers.ts:72]:
```suggestion
```

**Pull request merged**